Mixpanel/ReadWriteLock.swift

```diff
@@ -0,0 +1,31 @@
+//
+//  ReadWriteLock.swift
+//  Mixpanel
+//
+//  Created by Hairuo Sang on 8/9/17.
+//  Copyright © 2017 Mixpanel. All rights reserved.
+//
+
+import Foundation
+
+class ReadWriteLock{
```

Comment:
can you create a unit test for this? just to guarantee that it works the way we want